Client/Frontend/Browser/BrowserToolbar.swift

```diff
@@ -36,102 +30,72 @@ class BrowserToolbar: UIView, UITextFieldDelegate, BrowserLocationViewDelegate {
 
     override init(frame: CGRect) {
         super.init(frame: frame)
-        viewDidInit()
+        initViews()
     }
 
     required init(coder aDecoder: NSCoder) {
         super.init(coder: aDecoder)
-        viewDidInit()
     }
 
-    private func viewDidInit() {
-        self.backgroundColor = UIColor(white: 0.80, alpha: 1.0)
+    func addButtons(buttons: UIButton...) {
+        for button in buttons {
+            button.setTitleColor(UIColor.blackColor(), forState: UIControlState.Normal)
+            button.setTitleColor(UIColor.grayColor(), forState: UIControlState.Disabled)
+            button.imageView?.contentMode = UIViewContentMode.ScaleAspectFit
+            addSubview(button)
+        }
+
+        var prev: UIView? = nil
+        for view in self.subviews {
```

Comment:
This is very error prone. Who guarantees that the buttons are the only thing added to the view? Would it not be better to move the `addView` inside `addButtons` and loop over the array of buttons instead of all the subviews?